### ISSUE TITLE
Off topic

### ISSUE THREAD

**Issue by easleycorey** [Hardcore-fs/TinyWebServer#3]
Xyratex/Compellent
Hello greetings,

I'm trying to flash some disk shelves and wondering if you could help me out.

I have a few firmwares I found and some  controllers I have already work with my netapp ds4243s.  However I am getting very limited commanded over serial than you or others I have seen. Wondering if you could help enlighten me.

Sorry for awkward communication, this was the only place I could find where you were still registered

Thanks,
Corey
easleycorey@gmail.com

**Comment by Hardcore-fs** [comment]
Yep... someone took my whole site server out.

Never worked with the netapp.
Basically most of the disk shelves were built to a standard layout.
The firmware is specific to the manufacturer and as such they all had their own way of getting into it...

**Comment by easleycorey** [comment]
Holy cannoli you replied. The backplanes I'm running are netapp but the controller in each of the disk cabinets is Xyratex /Dell Compellent HB-SBB2-E601-COMP. It looks like the upgraded version of the one you have and wondering if I could have some help troubleshooting or at least a manual or something that might help me out. What do you mean someone took it down, the result wasn't your doing ? Do you pay for hosting and it got hacked? 
Thanks,
Corey

**Comment by Hardcore-fs** [comment]
yep.. it was that wordpress garbage again.... someone took the whole website down & corrupted the content.
there are no manuals, other than physical installation, which is not what you are looking for.

The controllers are stock...BUT.....
the software images they run are NOT.
starting with the boot image some are heavily modified. with massive sections of no-opp ed out code.
also they are subverted to hide commands, plus some commands are just written over with trash.

so basically the hardware can look identical, but the firmware is as different as linux is to windows.

a good starting point, is to take some of the upgrade images for upgrading, then drop them into a hex editor to find commands.
but even then some are hidden... with single byte compares instead of strings.

**Comment by easleycorey** [comment]
whelp finally sourced an update file. I wasn't able to find any way of updating it but found alot of other commands to help dump the firmware. kinda pointless if I can't find a way to flash what I want tho. Wonder if you might take a look? Please and Thank You. fine is a gff file.
[canister_local_combined.txt](https://github.com/Hardcore-fs/TinyWebServer/files/8633105/canister_local_combined.txt)

**Comment by Hardcore-fs** [comment]
that does not seem to be valid code

it normally starts like this

![k1](https://user-images.githubusercontent.com/1309004/167030709-dbb63349-c486-4df7-a1dc-5fa7eba4a610.jpg)

**Comment by easleycorey** [comment]
[Extracted_LaCie_Release_3.3_UIDBuild_0303000017_UID.zip](https://github.com/Hardcore-fs/TinyWebServer/files/8665903/Extracted_LaCie_Release_3.3_UIDBuild_0303000017_UID.zip)

Nope that .gff file is correct it flashes with a fwdownload tool.  I can now flash the canister just fine but on some of the "mid" which is the backplane need flashed and the dell windows update files did not includ the mid files. This is a zip of Lacie and a txt file noting all the gff files inside of the bin file. I tired using binwalk and some files and a elf file were extracted but near the number noted in UIDM-R3.3-ELCD-A6500F-0017-Contents.txt. Wondering what you can make out of all of it. To clarify the gff files like the one I uploaded before are the correct firmware files and are flashed over sas via ses pages. Thanks

**Comment by Hardcore-fs** [comment]
Sorry... but this is 6500 NOT 5400, they are completely different models and internal systems.
their load system is done completely differently as sub modules.

all the code i have is related to 5400/5412
